gammapy/cube/models.py

```diff
@@ -437,3 +438,49 @@ def compute_npred(self):
         if self.background:
             npred.data += self.background.data
         return npred.data
+
+
+class SkyMap3d(object):
+    """Cube sky map template model.
+
+    Parameters
+    ----------
+    map : `~gammapy.map.Map`
+        Map template
+    norm : `~astropy.units.Quantity`
+        Norm parameter (multiplied with map values)
+    meta : dict, optional
+        Meta information, meta['filename'] will be used for serialization
+    """
+
+    def __init__(self, map, norm=1, meta=None):
+        self._map = map
+        self.parameters = ParameterList([
+            Parameter('norm', norm),
+        ])
+        self.meta = dict() if meta is None else meta
+
+    @classmethod
+    def read(cls, filename, **kwargs):
+        """Read spatial template model from FITS image.
+
+
+        Parameters
+        ----------
+        filename : str
+            FITS image filename.
+        """
+        template = Map.read(filename, **kwargs)
+        return cls(template)
+
+    def evaluate(self, lon, lat, energy):
+        coord = dict(
+            lon=lon.to('deg').value,
+            lat=lat.to('deg').value,
+            energy=energy.value,
+        )
+        print(coord)
+        val = self._map.interp_by_coord(coord, fill_value=0)
+        norm = self.parameters['norm'].value
+        # TODO: use map unit? self._map.unit
+        return norm * val * u.Unit('sr-1')
```

Comment:
You can use the `MapGeom.solid_angle()`, especially if the norm is a flux unit.
 It can probably be stored as a member of the class itself.

Reply:
The decision about what to put here exactly in `evaluate` is coupled to what we do in
http://docs.gammapy.org/dev/api/gammapy.cube.MapEvaluator.html

I think at the moment, we already apply `solid_angle` and energy integration there, so here in the diffuse 3D model evaluate, you don't have to do anything apart from interpolating the map value. However, I think the unit is currently incorrect, you could either do this:
```
return norm * val * u.Unit('cm-2 s-1 sr-1 MeV-1')
```
hardcoding to the Fermi-LAT diffuse maps, or if they set the unit correctly in the map header, you could try this:
```
return norm * val * u.Unit(self._map.unit)
```
You could test this by taking one run from CTA DC1 and computing an npred cube using the SkyModelEvaluator. You expect ~ 1 count, if it's 1e6 or 1e-6 there's a unit problem.